File: apps/CounselCompanion/android/app/src/main/java/com/yeonseok/melangecounsel/ui/screens/DiagnosticsScreen.kt
package com.yeonseok.melangecounsel.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.yeonseok.melangecounsel.ui.viewmodel.CounselUiState
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

@Composable
fun DiagnosticsScreen(state: CounselUiState) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 16.dp)
            .padding(top = 16.dp),
        verticalArrangement = Arrangement.spacedBy(14.dp)
    ) {
        Text(
            text = "Diagnostics",
            style = MaterialTheme.typography.titleLarge,
            color = MaterialTheme.colorScheme.onSurface
        )

        DiagCard(title = "Model Info") {
            InfoRow("Model ID", state.modelId)
            InfoRow("Personal Key", state.maskedPersonalKey)
            InfoRow("Key Status", if (state.maskedPersonalKey.length > 8) "Configured" else "Missing")
            InfoRow("Stop Reason", state.diagnostics.lastStopReason)
        }

        DiagCard(title = "Last Generation") {
            InfoRow("Timestamp", formatTimestamp(state.diagnostics.lastRunTimestamp))
            InfoRow("Duration", "${state.diagnostics.lastGenerationMs} ms")
            InfoRow("Tokens", "${state.diagnostics.lastTokenCount}")
        }

        DiagCard(title = "Raw Log") {
            Text(
                text = state.diagnostics.lastRawLog.ifBlank { "No generation yet." },
                style = MaterialTheme.typography.bodyMedium.copy(
                    fontFamily = FontFamily.Monospace
                ),
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }

        Spacer(modifier = Modifier.height(16.dp))
    }
}

@Composable
private fun DiagCard(title: String, content: @Composable () -> Unit) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(18.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceContainerLowest
        )
    ) {
        Column(
            modifier = Modifier.padding(18.dp),
            verticalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            Text(
                text = title,
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Medium
            )
            HorizontalDivider(color = MaterialTheme.colorScheme.outlineVariant)
            content()
        }
    }
}

@Composable
private fun InfoRow(label: String, value: String) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Text(
            text = label,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant
        )
        Text(
            text = value,
            style = MaterialTheme.typography.bodyMedium,
            fontWeight = FontWeight.Medium,
            color = MaterialTheme.colorScheme.onSurface
        )
    }
}

private fun formatTimestamp(timestamp: Long): String {
    if (timestamp == 0L) return "No run yet"
    val formatter = SimpleDateFormat("MMM d, yyyy h:mm:ss a", Locale.getDefault())
    return formatter.format(Date(timestamp))
}

File: apps/CounselCompanion/ios/CounselCompanion/Views/DiagnosticsScreen.swift
import SwiftUI

struct DiagnosticsScreen: View {
    @EnvironmentObject var vm: ChatViewModel
    private let formatter: DateFormatter = {
        let f = DateFormatter(); f.dateStyle = .medium; f.timeStyle = .medium; return f
    }()

    var body: some View {
        ScrollView {
            VStack(alignment: .leading, spacing: 12) {
                Text("Diagnostics")
                    .font(.system(size: 17, weight: .semibold, design: .serif))
                    .padding(.horizontal, 16)
                    .padding(.top, 8)

                diagCard("Model Info") {
                    infoRow("Model ID", ZeticChatEngine.modelId)
                    Divider()
                    infoRow("Personal Key", ZeticChatEngine.maskedKey())
                    Divider()
                    infoRow("Key Status", ZeticChatEngine.maskedKey().count > 8 ? "Configured" : "Missing")
                }

                diagCard("Last Generation") {
                    infoRow("Timestamp", vm.diagnostics.lastRun.map { formatter.string(from: $0) } ?? "No run yet")
                    Divider()
                    infoRow("Duration", "\(vm.diagnostics.lastDurationMs) ms")
                    Divider()
                    infoRow("Tokens", "\(vm.diagnostics.lastTokenCount)")
                    Divider()
                    infoRow("Stop Reason", vm.diagnostics.lastStopReason)
                }

                diagCard("Raw Log") {
                    Text(vm.diagnostics.lastRawLog.isEmpty ? "No generation yet." : vm.diagnostics.lastRawLog)
                        .font(.system(size: 10, design: .monospaced))
                        .foregroundStyle(.secondary)
                        .textSelection(.enabled)
                }
            }
            .padding(.bottom, 24)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(
            LinearGradient(colors: [Color.bgTop, Color.bgBottom], startPoint: .top, endPoint: .bottom)
                .ignoresSafeArea(.container)
        )
    }

    private func diagCard(_ title: String, @ViewBuilder content: () -> some View) -> some View {
        VStack(alignment: .leading, spacing: 8) {
            Text(title)
                .font(.system(size: 13, weight: .medium))
            content()
        }
        .padding(14)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            RoundedRectangle(cornerRadius: 14, style: .continuous)
                .fill(Color(.tertiarySystemBackground))
        )
        .padding(.horizontal, 14)
    }

    private func infoRow(_ label: String, _ value: String) -> some View {
        HStack {
            Text(label)
                .font(.system(size: 11))
                .foregroundStyle(.secondary)
            Spacer()
            Text(value)
                .font(.system(size: 11, weight: .medium))
        }
    }
}
